src/time/rfc1123.cc

```diff
@@ -44,7 +44,13 @@ make_month(const char *s)
     char month[3];
 
     month[0] = xtoupper(*s);
+    if (!month[0])
+        return -1; // protects *(s + 1) below
```

Comment:
The new safety checks do not alter Squid behavior: Any (unlikely) overread side effects aside, we continue to accept valid 3-letter months and reject everything else AFAICT.